### PR TITLE
fix: omit https/2 pseudo headers

### LINKED ISSUE #421: http2 is failing with
```
capturing exception: could not handle request (058f571aa3784fe995c9e28941c65312) TypeError: Headers.append: ":method" is an invalid header name.
    at webidl.errors.exception (/Users/punkpeye/Developer/punkpeye/glama/node_modules/.pnpm/undici@6.19.8/node_modules/undici/lib/web/fetch/webidl.js:13:10)
    at webidl.errors.invalidArgument (/Users/punkpeye/Developer/punkpeye/glama/node_modules/.pnpm/undici@6.19.8/node_modules/undici/lib/web/fetch/webidl.js:29:24)
    at appendHeader (/Users/punkpeye/Developer/punkpeye/glama/node_modules/.pnpm/undici@6.19.8/node_modules/undici/lib/web/fetch/headers.js:89:25)
    at Headers.append (/Users/punkpeye/Developer/punkpeye/glama/node_modules/.pnpm/undici@6.19.8/node_modules/undici/lib/web/fetch/headers.js:390:12)
```

If I enable http2 in fastify, then the requests fail with the above error.

### PR DESCRIPTION
**Pull request by mcansh**
closes #421

### CODE REVIEW

File: packages/remix-fastify/src/server.ts [hunk 1]
```diff
@@ -71,6 +71,7 @@ export function createRemixHeaders(
   let headers = new Headers();
 
   for (let [key, values] of Object.entries(requestHeaders)) {
+    if (key.startsWith(":")) continue;
```
Comment (punkpeye) [inline packages/remix-fastify/src/server.ts:74]: I did need to add `host` from `:authority` though

**Comment by punkpeye** [comment]
I saw you closed this. Did you surface contrary to what I have?

**Comment by mcansh** [comment]
> I saw you closed this. Did you surface contrary to what I have?

was a resolved upstream issue in Remix https://github.com/remix-run/remix/issues/7867

**Comment by punkpeye** [comment]
thanks for letting me know

**Pull request closed without merging**